drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -4238,6 +4241,47 @@ static struct notifier_block netevent_notifier = {
 	.notifier_call = rswitch_netevent_cb,
 };
 
+static ssize_t l3_offload_show(struct device *dev, struct device_attribute *attr,
+			       char *buf)
+{
+	return sysfs_emit(buf, "%d\n", glob_priv->offload_enabled);
+}
+
+static void rswitch_disable_offload(struct rswitch_private *priv)
+{
+	struct rswitch_device *rdev;
+
+	read_lock(&priv->rdev_list_lock);
+	list_for_each_entry(rdev, &priv->rdev_list, list)
+		cleanup_all_routes(rdev);
+	read_unlock(&priv->rdev_list_lock);
+}
+
+static ssize_t l3_offload_store(struct device *dev, struct device_attribute *attr,
+				const char *buf, size_t count)
+{
+	long new_value;
+
+	if (kstrtol(buf, 10, &new_value))
+		return -EINVAL;
+
+	new_value = !!new_value;
+	if (new_value != glob_priv->offload_enabled) {
+		if (new_value) {
+			register_fib_notifier(&init_net, &glob_priv->fib_nb, NULL, NULL);
+		} else {
+			unregister_fib_notifier(&init_net, &glob_priv->fib_nb);
+			rswitch_disable_offload(glob_priv);
+		}
+
+		glob_priv->offload_enabled = new_value;
+	}
+
+	return count;
+}
+
+static DEVICE_ATTR(l3_offload, 0644, l3_offload_show, l3_offload_store);
```

Comment:
`s/0644/S_IRUGO | S_IWUSR`

Reply:
I changed it according to checkpatch recommendations to DEVICE_ATTR_RW. It's equivalent to S_IRUGO | S_IWUSR or 0644.